### PR TITLE
fix(builtin-function): joining() function renamed and aligned with join

### LINKED ISSUE #372: Align with new string join() built-in function
**Is your feature request related to a problem? Please describe.**
A new built-in function `joining()` was implemented with #339. In the meantime, a new built-in function `string join()` was defined for DMN 1.4 that provides the same logic. 
 

**Describe the solution you'd like**
Since the new function is not available in a released version yet, we should rename the function from `joining()` to `string join()` to align with the specs. 

image

Open TODOs: 
* rename function to `string join()`
* ignore `null` items in the list
* handle `null` as empty `delimiter` 
* write documentation

**Related issues**

* Camunda BPM: 
* Zeebe broker: 
* OMG: https://issues.omg.org/browse/DMN14-116

### PR DESCRIPTION
**Pull request by P3trur0**
## Description

* joining function renamed as join
* join function behavior modified to reflect specs (e.g.: empty string and null delimiter management)
* unit tests added
* documentation updated

## Related issues

closes #372

### CODE REVIEW

File: src/main/scala/org/camunda/feel/impl/builtin/ListBuiltinFunctions.scala [hunk 1]
```diff
@@ -43,9 +43,9 @@ object ListBuiltinFunctions {
     "distinct values" -> List(distinctValuesFunction),
     "flatten" -> List(flattenFunction),
     "sort" -> List(sortFunction),
-    "joining" -> List(joiningFunction,
-                      joiningWithDelimiterFunction,
-                      joiningWithDelimiterAndPrefixAndSuffixFunction)
+    "join" -> List(joinFunction,
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/builtin/ListBuiltinFunctions.scala:46]: :x: We must rename the function to `string join`. In the spec, the function is defined as `string join()`. It is a new practice in the spec to use the data type as prefix :sweat_smile: 

```suggestion
    "string join" -> List(joinFunction,
```

**Review by saig0 — CHANGES_REQUESTED** [review]
@P3trur0 thank you for your contribution :rocket: 

It looks really good :+1: Just one point that must be changed. Please have a look at my comment.

File: src/main/scala/org/camunda/feel/impl/builtin/ListBuiltinFunctions.scala (hunk 1 above)
Comment (P3trur0) [reply src/main/scala/org/camunda/feel/impl/builtin/ListBuiltinFunctions.scala:46]: oh, sorry, I missed that! I am going to fix it.

**Comment by P3trur0** [comment]
> @P3trur0 thank you for your contribution 🚀
> 
> It looks really good 👍 Just one point that must be changed. Please have a look at my comment.

fixed. Please, have a look!

**Review by saig0 — APPROVED** [review]
@P3trur0 super :tada: Thank you :cake:

**Pull request merged**